packages/react-shadow/src/util.ts

```diff
@@ -0,0 +1,12 @@
+import { REACT_SHADOW_INSERTION_BUCKET_NAME } from './constants';
+import type { GriffelShadowDOMRenderer } from '@griffel/shadow-dom';
+
+// Hack to get this type that is not directly exported at the moment.
+type ExtendedCSSStyleSheet = GriffelShadowDOMRenderer['adoptedStyleSheets'][0];
+
+export const makeInsertionPointSheet = (): ExtendedCSSStyleSheet => {
```

Comment:
since app CSSStylesheets need to be created, why is it necessary to make the user create a new sentinel stylesheet? couldn't they simply pass the last stylesheet they append as the insertion point?

You could then find the insertion point by a simple equality check without any extra metadata. It's also easier in userland since they don't need to call any specific utility to make this work

Reply:
I like this idea (it's where I originally started), but I couldn't see a good way to pass the sentinal into `root`. Will consider it more.

Reply:
With the current `createProxy()` API I don't think there is a way to do this unfortunately. Perhaps @layershifter has an idea though?